samza-core/src/main/java/org/apache/samza/container/ContainerHeartbeatMonitor.java

```diff
@@ -170,4 +212,55 @@ public Thread newThread(Runnable runnable) {
       return t;
     }
   }
+
+  static final class ContainerHeartbeatMetrics {
+    private static final String GROUP = "ContainerHeartbeatMonitor";
+    private static final String HEARTBEAT_DISCOVERY_TIME_MS = "heartbeat-discovery-time-ms";
+    private static final String HEARTBEAT_ESTABLISHED_FAILURE_COUNT = "heartbeat-established-failure-count";
+    private static final String HEARTBEAT_ESTABLISHED_WITH_NEW_AM_COUNT = "heartbeat-established-with-new-am-count";
+    private static final String HEARTBEAT_EXPIRED_COUNT = "heartbeat-expired-count";
+
+    private final Counter heartbeatEstablishFailureCount;
```

Comment:
nit - Change `heartbeatEstablishFailureCount` to `heartbeatEstablishedFailureCount`